### ISSUE TITLE
Idea: Providing the assemblies built by the solution as gulp output stream

### ISSUE THREAD

**Issue by redwards1966** [hoffi/gulp-msbuild#14]
I know this is possible using an MSBuild targets file but not sure if you can get this from the msbuild exe or not somehow.

In my existing targets file (the one I'm hoping to eliminate by using gulp) I have the following for my build task:

	<Target Name="Build">
		<Message Text="Building projects ... " />
		<MSBuild Projects="@(ProjectToBuild)" Targets="Build" Properties='DefineConstants=DINVERSION="$(DINVersion)"'>
		    <Output
				TaskParameter="TargetOutputs"
				ItemName="AssembliesBuiltBySolution" />
		</MSBuild>
	</Target>

Basically at the end of the build, I get a named output 'AssembliesBuiltBySolution' that I can then copy to another path using this in another target:

    
    <Copy SourceFiles="@(AssembliesBuiltBySolution)" DestinationFolder="$(TempPath)"  />

I am thinking that in 'gulp thinking', I should be able to just pipe that file list to a destination and that would be it.

Is this even possible do you think?

**Comment by hoffi** [comment]
Hm im not sure, but i think it is not possible, as i don't know how to figure out, where the built assemblies are saved, as you still can specify a non-default Output-Path in the Project file for example.

But maybe this helps you:
You can specify a different Output-Path with a custom property (e.g. `/property:OutDir=[TEMP_PATH]`)

**Comment by redwards1966** [comment]
Yeah, I ended up using properties: { OutputPath: config.binPath } and that
threw everything into the path I needed.
Thanks again!


On Tue, Mar 3, 2015 at 12:31 PM, Stefan Hoffmann <notifications@github.com>
wrote:

> Hm im not sure, but i think it is not possible, as i don't know how to
> figure out, where the built assemblies are saved, as you still can specify
> a non-default Output-Path in the Project file for example.
>
> But maybe this helps you:
> You can specify a different Output-Path with a custom property (e.g.
> /property:OutDir=[TEMP_PATH])
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hoffi/gulp-msbuild/issues/14#issuecomment-76992936>.
>



-- 
Regards,

Richard

**Comment by hoffi** [comment]
No problem :)
